src/Sarif/Visitors/FilteringVisitor.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis.Sarif.Writers;
+
+namespace Microsoft.CodeAnalysis.Sarif.Visitors
+{
+
+    /// <summary>
+    /// A visitor that filters a specified SARIF log to create a new log containing only those
+    /// results for which a specified predicate returns true, and only those elements of run-level
+    /// collections such as Run.Artifacts that are relevant to the filtered results.
+    /// </summary>
+    public class FilteringVisitor : SarifRewritingVisitor
+    {
+        public delegate bool FilteringPredicate(Result result);
+
+        private readonly FilteringPredicate predicate;
```

Comment:
Maybe rename to "IncludeResult" so the desired return value to include a result is clearer?

Reply:
Ah, you're saying that somebody might think this predicate should return true if you want to _filter out_ the result. Ok, I'll rename it to `IncludedResultPredicate`.

---
In reply to: [338198865](https://github.com/microsoft/sarif-sdk/pull/1708#discussion_r338198865) [](ancestors = 338198865)